Sources/Caching/PostedTransactionCache.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PostedTransactionCache.swift
+//
+//  Created by Nacho Soto on 7/27/23.
+
+import Foundation
+
+protocol PostedTransactionCacheType {
+
+    func savePostedTransaction(_ transaction: StoreTransactionType)
+    func hasPostedTransaction(_ transaction: StoreTransactionType) -> Bool
+
+}
+
+// TODO: tests
+// TODO: integrate
+
+final class PostedTransactionCache: PostedTransactionCacheType {
```

Comment:
This uses `DeviceCache` underneath (composition, not inheritance). It also doesn't make `DeviceCache` have any more domain-specific caching code.

This is how I want to refactor `DeviceCache` so it's split in smaller classes.